File: python/examples/streaming_chat.py
"""Streaming chat, sync and async — the same request, two ways to read it.

INFERHUB_BASE=http://localhost:5080/ INFERHUB_API_KEY=... python examples/streaming_chat.py
"""

import asyncio
import os

from inferhub_client import (
    AsyncInferHubClient,
    ChatMessage,
    ChatRequest,
    InferHubClient,
)

base_url = os.environ.get("INFERHUB_BASE", "http://localhost:5080/")
api_key = os.environ.get("INFERHUB_API_KEY")


def sync_stream() -> None:
    with InferHubClient(base_url, api_key) as client:
        request = ChatRequest(
            model="llama3", messages=[ChatMessage("user", "Stream me a haiku.")]
        )
        for chunk in client.chat_stream(request):
            print(chunk.message.content or "", end="", flush=True)
        print()


async def async_stream() -> None:
    async with AsyncInferHubClient(base_url, api_key) as client:
        request = ChatRequest(
            model="llama3", messages=[ChatMessage("user", "Stream me a haiku.")]
        )
        async for chunk in client.chat_stream(request):
            print(chunk.message.content or "", end="", flush=True)
        print()


if __name__ == "__main__":
    sync_stream()
    asyncio.run(async_stream())

File: js/examples/streaming-chat.ts
/**
 * Streaming chat — one ChatResponse per NDJSON line, printed as it arrives.
 *
 * INFERHUB_BASE=http://localhost:5080/ INFERHUB_API_KEY=... npx tsx examples/streaming-chat.ts
 */

import { InferHubClient, InferHubError } from "../src/index.js";

const baseUrl = process.env.INFERHUB_BASE ?? "http://localhost:5080/";
const apiKey = process.env.INFERHUB_API_KEY;

const client = new InferHubClient({ baseUrl, apiKey });

try {
  for await (const chunk of client.chatStream({
    model: "llama3",
    messages: [{ role: "user", content: "Count from one to five." }],
  })) {
    process.stdout.write(chunk.message?.content ?? "");
    if (chunk.done) {
      console.log(`\n\n(served by: ${chunk.servedBy ?? "unknown"})`);
    }
  }
} catch (error) {
  // A mid-stream terminal error ({"error": ..., "done": true}) throws InferHubError instead of
  // the iterator hanging or ending quietly with a partial answer nobody was told about.
  if (error instanceof InferHubError) {
    console.error(`\nstream terminated: ${error.message}`);
  } else {
    throw error;
  }
}
